src/relay/connection.ts

```diff
@@ -358,6 +360,8 @@ class RelayConnection extends EventEmitter implements MultiaddrConnection {
       if (received.done) {
         currentSource = undefined
         streamPromise = undefined
+        this.destroyed = true
+        this._destroyedPromise.resolve()
```

Comment:
RelayConnection is going to wait for the next connection, so I don't think we should mark the connection destroyed.